src/image_cache/directory.rs

```diff
@@ -215,8 +215,9 @@ impl Directory {
 	}
 
 	pub fn update_directory(&mut self) -> Result<()> {
-		let curr_filename = self.curr_filename();
+		let (curr_filename, curr_index) = (self.curr_filename(), self.curr_file_idx);
 		self.collect_directory()?;
+		// if possible preserve previous file_name
```

Comment:
I'm guessing you meant to put this right before the next loop as it is out of place here.